modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj

```diff
@@ -123,27 +124,34 @@
 (defn- format-env-key [env-key]
   (let [[_ header body footer]
         (re-find #"(-----BEGIN (?:\p{Alnum}+ )?PRIVATE KEY-----)(.*)(-----END (?:\p{Alnum}+ )?PRIVATE KEY-----)" env-key)]
-    (str header (str/replace body #"\s+" "\n") footer)))
+    (str header (str/replace body #"\s+|\\n" "\n") footer)))
 
 (deftest can-connect-test
-  (mt/test-driver :snowflake
-    (let [can-connect? (partial driver/can-connect? :snowflake)]
-      (is (= true
-             (can-connect? (:details (mt/db))))
-          "can-connect? should return true for normal Snowflake DB details")
-      (is (thrown?
-           net.snowflake.client.jdbc.SnowflakeSQLException
-           (can-connect? (assoc (:details (mt/db)) :db (mt/random-name))))
-          "can-connect? should throw for Snowflake databases that don't exist (#9511)")
-      (let [pk-user (tx/db-test-env-var-or-throw :snowflake :pk-user)
-            pk-key  (format-env-key (tx/db-test-env-var-or-throw :snowflake :pk-private-key))]
-        (is (= true
-               (-> (:details (mt/db))
-                   (dissoc :password)
-                   (assoc :user pk-user
-                          :private-key-value pk-key)
-                   can-connect?))
-            "can-connect? should return true when authenticating with private key")))))
+  (let [pk-key (format-env-key (tx/db-test-env-var-or-throw :snowflake :pk-private-key))
+        pk-user (tx/db-test-env-var :snowflake :pk-user)
+        pk-db (tx/db-test-env-var :snowflake :pk-db "SNOWFLAKE_SAMPLE_DATA")]
+    (mt/test-driver :snowflake
+      (mt/with-temp-file [pk-path]
+        (let [can-connect? (partial driver/can-connect? :snowflake)]
+          (is (= true
+                 (can-connect? (:details (mt/db))))
+              "can-connect? should return true for normal Snowflake DB details")
+          (is (thrown?
+               net.snowflake.client.jdbc.SnowflakeSQLException
+               (can-connect? (assoc (:details (mt/db)) :db (mt/random-name))))
+              "can-connect? should throw for Snowflake databases that don't exist (#9511)")
+          (when (and pk-key pk-user)
```

Comment:
When would the key and user env vars not be set?

Reply:
Those will be nil whenever the env variables `MB_SNOWFLAKE_TEST_PK_PRIVATE_KEY` and `MB_SNOWFLAKE_TEST_PK_USER` are not set

Reply:
So, if you were to run snowflake tests without wanting to test the key pair auth (which requires a lot of setup), then this test will still pass